tests/test_faster_rcnn.py

```diff
@@ -0,0 +1,377 @@
+
+    assert os.listdir("/tmp/load/")[0].startswith("events.out.tfevents")
+
+
+@patch("datasetinsights.estimators.faster_rcnn.Dataset.create")
+def test_download_data(mock_create, config):
+    mock_create.return_value = dummy_data
+    train_dataset = download_data(config, TRAIN)
+    assert len(dummy_data.images) == len(train_dataset)
+
+
+@patch("datasetinsights.estimators.faster_rcnn.Dataset.create")
+def test_create_dryrun_dataset(mock_create, config):
+    mock_create.return_value = dummy_data
+    train_dataset = download_data(config, TRAIN)
+    train_dataset = create_dryrun_dataset(config, train_dataset, TRAIN)
+    assert config.train.batch_size * 2 == len(train_dataset)
+
+
+@patch("datasetinsights.estimators.faster_rcnn.Dataset.create")
+def test_create_sampler(mock_create, config):
+    mock_create.return_value = dummy_data
+    train_dataset = download_data(config, TRAIN)
+    is_distributed = config.system.distributed
+    train_sampler = create_sampler(
+        is_distributed=is_distributed, dataset=train_dataset, is_train=True
+    )
+    assert len(dummy_data.images) == len(train_sampler)
+
+
+@patch("datasetinsights.estimators.faster_rcnn.Dataset.create")
+def test_create_dataloader(mock_create, config):
+    mock_create.return_value = dummy_data
+    train_dataset = download_data(config, TRAIN)
+    is_distributed = config.system.distributed
+    train_sampler = create_sampler(
+        is_distributed=is_distributed, dataset=train_dataset, is_train=True
+    )
+    train_loader = create_dataloader(
+        config, train_dataset, train_sampler, TRAIN
+    )
+    assert train_loader
+
+
+@patch("datasetinsights.estimators.faster_rcnn.Dataset.create")
+def test_load_data(mock_create, config):
+    mock_create.return_value = dummy_data
+    train_dataset = download_data(config, TRAIN)
+    is_distributed = config.system.distributed
+    train_sampler = create_sampler(
+        is_distributed=is_distributed, dataset=train_dataset, is_train=True
+    )
+    dataloader = load_data(
+        config=config,
+        dataset=train_dataset,
+        batch_size=config.train.batch_size,
+        sampler=train_sampler,
+        collate_fn=collate_fn,
+        train=True,
+    )
+    assert dataloader
+
+
+@patch("datasetinsights.estimators.faster_rcnn.Dataset.create")
+def test_create_optimizer(mock_create, config):
+    mock_create.return_value = dummy_data
+    config.system.logdir = "/tmp/train/"
+    kfp_writer = MagicMock()
+    writer = SummaryWriter(config.system.logdir, write_to_disk=True)
+    checkpointer = create_checkpointer(logdir=writer.logdir, config=config)
+    estimator = Estimator.create(
+        config.estimator,
+        config=config,
+        writer=writer,
+        device=torch.device("cpu"),
+        checkpointer=checkpointer,
+        kfp_writer=kfp_writer,
+    )
+    optimizer, lr_scheduler = create_optimizer_lrs(config, estimator.model)
+
+    assert optimizer
```

Comment:
What's the purpose of this assert? It will always be true if the object exists. I think a better test that more robust to external API change can be:

```
def test_optimizer_lr():
    # mock torch.optim.Adam
    # mock torch.optim.lr_scheduler.LambdaLR
    config = {...name: Adam, args: {...}}
    op, lr = create_optimizer_lrs(config, params)
    
    assert mock_adam == op
    assert mock_lr = lr
    op.assert_called_with(...)
```

Does this pattern applies to other static methods or module level methods?